addon/utils/group-utils.js

```diff
@@ -80,7 +80,7 @@ export function filterOptions(options, text, matcher, skipDisabled = false) {
     if (isGroup(entry)) {
       let suboptions = filterOptions(get(entry, 'options'), text, matcher, skipDisabled);
       if (get(suboptions, 'length') > 0) {
-        opts.push({ groupName: entry.groupName, options: suboptions });
```

Comment:
I get it now. During filtering, a new objects is created.

I'd only suggest one change `{ groupName: entry.groupName, disabled: entry.disabled, options: suboptions }`. The generated group should have the same value as the original, including undefined. There is no need of coercing it to true.